File: solutions/typescript/armstrong-numbers/1/armstrong-numbers.ts
export function isArmstrongNumber(num: number | bigint): boolean {
    let sum = BigInt(0);
    const numStr = num.toString();
    const len = BigInt(numStr.length);

    for (let i = 0; i < numStr.length; i++) {
        const digit = BigInt(numStr[i]);
        sum += digit ** len;
    }

    return sum === BigInt(num);
}

File: solutions/lua/armstrong-numbers/1/armstrong-numbers.lua
local ArmstrongNumbers = {}

function ArmstrongNumbers.is_armstrong_number(number)
    local sum = 0
    local len = string.len(number)
    for i = 1, len do
        sum = sum + tonumber(string.sub(number, i, i)) ^ len
    end
    return sum == tonumber(number)
end

return ArmstrongNumbers
